evalml/utils/gen_utils.py

```diff
@@ -353,11 +354,39 @@ def pad_with_nans(pd_data, num_to_pad):
         pd.DataFrame or pd.Series
     """
     if isinstance(pd_data, pd.Series):
-        padding = pd.Series([None] * num_to_pad)
+        padding = pd.Series([np.nan] * num_to_pad)
     else:
-        padding = pd.DataFrame({col: [None] * num_to_pad
+        padding = pd.DataFrame({col: [np.nan] * num_to_pad
                                 for col in pd_data.columns})
-    return pd.concat([padding, pd_data], ignore_index=True).infer_objects()
+    padded = pd.concat([padding, pd_data], ignore_index=True)
+    # By default, pd.concat will convert all types to object if there are mixed numerics and objects
+    # The call to convert_dtypes ensures numerics stay numerics in the new dataframe.
+    return padded.convert_dtypes(infer_objects=True, convert_string=False,
+                                 convert_integer=False, convert_boolean=False)
+
+
+def _get_rows_without_nans(*data):
+    """Compute a boolean array marking where all entries in the data are non-nan.
+
+    Arguments:
+        *data (sequence of pd.Series or pd.DataFrame)
+
+    Returns:
+        np.ndarray: mask where each entry is True if and only if all corresponding entries in data
+            are non-nan.
+    """
+    def _not_nan(pd_data):
+        if pd_data is None:
+            return np.array([True])
+        if isinstance(pd_data, pd.Series):
+            return ~pd_data.isna().values
+        elif isinstance(pd_data, pd.DataFrame):
+            return ~pd_data.isna().any(axis=1).values
+        else:
+            return pd_data
+
+    mask = reduce(lambda a, b: np.logical_and(_not_nan(a), _not_nan(b)), data)
```

Comment:
This is cool! So this means that if the second Dataframe/Series in the sequence had a nan value in the last column, then the last value in the mask will be False, like below? 
![image](https://user-images.githubusercontent.com/22552445/102283033-98702100-3eff-11eb-9268-d3c61e08e816.png)

Might be nice to update the return to `mask where each entry is True if and only if all corresponding entries in that index in data are non-nan` or something of the sort. I wasn't sure what this was supposed to do

Reply:
Yes exactly. Great suggestion on the docstring!